dev/com.ibm.ws.channelfw/src/com/ibm/ws/tcpchannel/internal/WorkQueueManager.java

```diff
@@ -100,6 +101,11 @@ protected WorkQueueManager() {
         }
     }
 
+    protected void startSelectors(boolean inBound, boolean startImmediately) throws ChannelException {
+        this.startImmediately = startImmediately;
+        this.startSelectors(inBound);
+    }
+
     protected void startSelectors(boolean inBound) throws ChannelException {
         if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) {
             Tr.entry(tc, "startSelectors " + inBound);
```

Comment:
should this class _always_ pass `_checkCancel` to the `SocketRWChannelSelector` ctor? There are a few instances in `startSelectors()` and `moveIntoPosition()` where we don't

Reply:
The intention is that the new `startSelector(boolean, boolean)` signature is used by outbound scenarios. When used it will set a class level `startImmediately` flag, such that when the original `startSelector(boolean)` is invoked, all constructors for outbound scenarios will inherit this `startImmediately` flag. All other constructors (inbound cases) should remain unchanged to preserve current behavior.

Reply:
Thanks for the explanation!